### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by zainm97**
## Description
This commit adds pagination to the app and pull in data from the state into the confirmation screen of the invitations 

https://nhsd-jira.digital.nhs.uk/browse/GAL-568
https://nhsd-jira.digital.nhs.uk/browse/GAL-556
https://nhsd-jira.digital.nhs.uk/browse/GAL-558

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/page.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default function Root() {
     <div>
       <body className="js-enabled">
         <AppStateProvider>
-          <Header />
+          {/* <Header /> */}
           <Start />
```
Comment (Amar-NHS) [inline src/app/page.jsx:19]: why have we commented out Header component in this file and importing them individually on the page's class component instead?

File: src/app/views/clinic_information/ClinicInformation.js [hunk 2]
```diff
@@ -119,14 +139,17 @@ class ClinicInformation extends Component {
     this.setState({
       appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetFillToInputValue / 100)),
     });
+    this.context.setState({
+      targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetFillToInputValue / 100))
+    })
```
Comment (Amar-NHS) [inline src/app/views/clinic_information/ClinicInformation.js:144]: This is what made me think if this branch was off 414 as I have made comment about this there. Will have to replace the local state variable with global one - not good practice to have duplicates.

File: src/app/views/clinic_information/ClinicInformation.js [hunk 3]
```diff
@@ -156,6 +179,9 @@ class ClinicInformation extends Component {
     this.setState({
       targetFillToInputValue: e.target.value,
     });
+    this.context.setState({
+      targetPercentageToFill: e.target.value
+    })
```
Comment (Amar-NHS) [inline src/app/views/clinic_information/ClinicInformation.js:184]: Will have to replace the local state variable with global one - not good practice to have duplicates.

File: src/app/views/clinic_information/ClinicInformation.js [hunk 4]
```diff
@@ -263,6 +289,15 @@ class ClinicInformation extends Component {
...
             displayViewAllPrevInvitations: displayViewAllPrevInvitations,
+            currentPage: 1,
+            pageSize: 10,
+          }, () => {
+            this.setState({
+              appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (this.state.targetFillToInputValue / 100)),
+            });
+            this.context.setState({
+              targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (this.state.targetFillToInputValue / 100))
+            })
```
Comment (Amar-NHS) [inline src/app/views/clinic_information/ClinicInformation.js:300]: same here, will have to replace the local state variable with global one - not good practice to have duplicates.

File: src/app/views/clinic_information/ClinicInformation.js [hunk 5]
```diff
@@ -350,16 +385,40 @@ class ClinicInformation extends Component {
...
+                  )
+                  .then((response) => {
+                    const targetPercentageValue = response.data.targetPercentage.N;
+                    this.setState({
+                      targetFillToInputValue: targetPercentageValue,
+                      appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetPercentageValue / 100)),
+                    });
+                    this.context.setState({
+                      targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetPercentageValue / 100)),
+                      targetPercentageToFill: targetPercentageValue
```
Comment (Amar-NHS) [inline src/app/views/clinic_information/ClinicInformation.js:407]: same here, will have to replace the local state variable with global one - not good practice to have duplicates.

File: src/app/views/clinic_information/LsoaTable.jsx [hunk 6]
```diff
@@ -1,188 +1,228 @@
-import React from "react";
+import React, { useState } from "react";
+import Pagination from "../../components/pagination";
+import { AppStateContext } from '@/app/context/AppStateContext';
```
Comment (Amar-NHS) [inline src/app/views/clinic_information/LsoaTable.jsx:3]: don't need this import as its not being used here

File: src/app/views/clinic_summary/ClinicSummary.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ import {
   daysSinceLastInvite,
 } from './helper';
 import axios from 'axios';
+import Header from "@/app/components/Header";
```
Comment (Amar-NHS) [inline src/app/views/clinic_summary/ClinicSummary.js:12]: Any reason, why Header component I seeing imported like this, compared to leaving it on the page.jsx? It looks like the same component file being used anyway/

File: src/app/views/invitation_planning/InvitationPlanning.js [hunk 8]
```diff
@@ -3,6 +3,7 @@ import { QuintileTarget } from "@/app/models/invitation_planning/QuintileTarget"
 import { sumQuintiles } from "./helper";
 import InvitationPlanningPage from "./InvitationPlanningPage";
 import axios from "axios";
+import Header from "@/app/components/Header";
```
Comment (Amar-NHS) [inline src/app/views/invitation_planning/InvitationPlanning.js:6]: same here with header being imported directly to the file

**Review by Amar-NHS — COMMENTED** [review]
Have left some comments to address, and also shared the screen recording to show other issues.

File: src/app/page.jsx (hunk 1 above)
Comment (zainm97) [reply src/app/page.jsx:19]: as discussed need to move these into respective components as menu bar shouldnt be accessible before login

File: src/app/views/clinic_information/LsoaTable.jsx (hunk 6 above)
Comment (zainm97) [reply src/app/views/clinic_information/LsoaTable.jsx:3]: good spot

File: src/app/views/clinic_summary/ClinicSummary.js (hunk 7 above)
Comment (zainm97) [reply src/app/views/clinic_summary/ClinicSummary.js:12]: as mentioned above

File: src/app/views/invitation_planning/InvitationPlanning.js (hunk 8 above)
Comment (zainm97) [reply src/app/views/invitation_planning/InvitationPlanning.js:6]: as mentioned above

File: src/app/views/clinic_information/ClinicInformation.js (hunk 2 above)
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js:144]: Will address these in algorithm ticket

File: src/app/views/clinic_information/ClinicInformation.js (hunk 4 above)
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js:300]: Will address these in algorithm ticket

File: src/app/views/clinic_information/ClinicInformation.js (hunk 5 above)
Comment (Amar-NHS) [reply src/app/views/clinic_information/ClinicInformation.js:407]: as you mentioned.

**Review by Amar-NHS — APPROVED** [review]
looks fine. As discussed the state variable duplicates, miles and per pages persistent issue will be addressed in another ticket.

**Review by Abdul-NHS — APPROVED** [review]
I noticed an issue with the default values for the lsoaTable miles and lsoas per page reverting back to original after calculate number to invite button was clicked and then click back. We've agreed this will be remediated in a separate story. Rest looks good!

**Pull request merged**